receiver/prometheusexecreceiver/receiver.go

```diff
@@ -0,0 +1,322 @@
+		promReceiverConfig: promReceiverConfig,
+		port:               config.Port,
+	}, nil
+}
+
+// Start creates the configs and calls the function that handles the prometheus_exec receiver
+func (per *prometheusExecReceiver) Start(ctx context.Context, host component.Host) error {
+	// Shutdown channel
+	per.shutdownCh = make(chan struct{})
+
+	// Start the process with the built config
+	go per.manageProcess(context.Background(), host)
+
+	return nil
+}
+
+// getPromReceiverConfig returns the Prometheus receiver config
+func getPromReceiverConfig(cfg *Config) *prometheusreceiver.Config {
+	scrapeConfig := &config.ScrapeConfig{}
+
+	scrapeConfig.ScrapeInterval = model.Duration(cfg.ScrapeInterval)
+	scrapeConfig.ScrapeTimeout = model.Duration(defaultScrapeTimeout)
+	scrapeConfig.Scheme = "http"
+	scrapeConfig.MetricsPath = defaultMetricsPath
+	scrapeConfig.JobName = extractName(cfg)
+	scrapeConfig.HonorLabels = false
+	scrapeConfig.HonorTimestamps = true
+
+	// Set the proper target by creating one target inside a single target group (this is how Prometheus wants its scrape config)
+	scrapeConfig.ServiceDiscoveryConfig = sdconfig.ServiceDiscoveryConfig{
+		StaticConfigs: []*targetgroup.Group{
+			{
+				Targets: []model.LabelSet{
+					{model.AddressLabel: model.LabelValue(fmt.Sprintf("localhost:%v", cfg.Port))},
+				},
+			},
+		},
+	}
+
+	receiverSettings := &configmodels.ReceiverSettings{}
+	receiverSettings.SetType(typeStr)
+	receiverSettings.SetName(cfg.Name())
+
+	return &prometheusreceiver.Config{
+		ReceiverSettings: *receiverSettings,
+		PrometheusConfig: &config.Config{
+			ScrapeConfigs: []*config.ScrapeConfig{scrapeConfig},
+		},
+	}
+}
+
+// getSubprocessConfig returns the subprocess config
+func getSubprocessConfig(cfg *Config) *subprocessmanager.SubprocessConfig {
+	subprocessConfig := &subprocessmanager.SubprocessConfig{}
+
+	subprocessConfig.Command = cfg.SubprocessConfig.Command
+	subprocessConfig.Env = cfg.SubprocessConfig.Env
+
+	return subprocessConfig
+}
+
+// extractName will return the receiver's given custom name (prometheus_exec/custom_name)
+func extractName(cfg *Config) string {
+	splitName := strings.SplitN(cfg.Name(), "/", 2)
+	if len(splitName) > 1 && splitName[1] != "" {
+		return splitName[1]
+	}
+	// fall back to the first part of the string, prometheus_exec
+	return splitName[0]
+}
+
+// manageProcess will put the process in an infinite starting loop which goes through the following steps
+// If the port is not defined by the user, one is generated and a new metrics receiver is built with the new port
+// All instances of {{port}} are replaced with the actual port (either defined by the user or generated)
+// Start the receiver and then the subprocess and get its runtime, then shutdown receiver since it crashed
+// Finally, the wait time before the subprocess is restarted is computed and this goroutine sleeps for that amount of time, before restarting the loop from the start
```

Comment:
^ Maybe you can abbreviate that comment now that that documentation is implied by the function calls below